packages/cxl-ui/src/components/cxl-marketing-nav.js

```diff
@@ -1,631 +1,339 @@
-
-        backBtn.classList.add('context-menu-item-back-button');
-        backBtn.innerHTML = '<vaadin-icon icon="lumo:angle-left"></vaadin-icon> Back';
-
-        menuItem.classList.add('back-button-menu-item');
-        menuItem.appendChild(backBtn);
-
-        menuItem.addEventListener('click', this._onBackBtnClick.bind(this));
-
-        // eslint-disable-next-line no-param-reassign
-        self[i] = { component: menuItem };
-      }
-    });
-
-    return contextMenuItems;
+  toogleSearch() {
+    this.searchDialogOpened = !this.searchDialogOpened;
   }
 
-  /**
-   * Highlight menu item with special class.
-   * Improves visual "Where am I?" navigation clarity.
-   *
-   * @since 2020.04.12
-   * @private
-   */
-  _highlightCurrentMenuItem() {
-    this.menuItemsElements.forEach((menuItemsEl) => {
-      const currentMenuItemEl = menuItemsEl.querySelector('.current-menu-item');
-
-      if (currentMenuItemEl && currentMenuItemEl.id && menuItemsEl.items) {
-        const idx = menuItemsEl.items.findIndex((i) => i.id === currentMenuItemEl.id);
-
-        menuItemsEl.setAttribute('selected', idx);
-      }
-    });
+  // eslint-disable-next-line class-methods-use-this
+  _searchDialogRenderer() {
+    return html` <div class="search-form-container"></div> `;
   }
 
-  /**
-   * Populate children `<vaadin-context-menu>` elements.
-   *
-   * @private
-   * @todo Links cannot be sub-menu hosts?
-   */
-  _updatedContextMenuItems() {
-    Object.values(this.contextMenuItems).forEach((items) => {
-      items.forEach((menuItem) => {
-        if (!menuItem.children || !menuItem.children.length) {
-          return;
-        }
-
-        const contextMenu = this.querySelector(
-          `vaadin-tab#menu-item-${menuItem.id} > vaadin-context-menu`
-        );
-
-        // Populate.
-        contextMenu.items = this._createContextMenuItems(menuItem.children);
-
-        // Prevent close on upstream events: clicks, keydown, etc
-        contextMenu.addEventListener('item-selected', (e) => {
-          e.stopImmediatePropagation();
-        });
-      });
-    });
+  // eslint-disable-next-line class-methods-use-this
+  _onBackBtnClick(e) {
+    e.stopImmediatePropagation();
+    [...document.body.querySelectorAll('vaadin-context-menu-overlay')].at(-1).close();
   }
 
-  /**
-   * Re-orient menu items.
-   *
-   * @see https://github.com/vaadin/vaadin-context-menu/blob/v4.3.12/src/vaadin-device-detector.html#L12
-   * @see https://github.com/vaadin/vaadin-context-menu/issues/253
-   * @todo refactor w/ `<cxl-icon-nav>`?
-   */
-  _reorientMenuItems() {
-    let orientation = 'vertical';
-
-    if (this.wide) {
-      orientation = 'horizontal';
+  // Creates and returns a custom menu item component for use in vaadin-menu-bar
+  createItem(
+    { text, description, sectionheader, component, icon, href, children, depth, split },
+    group
+  ) {
+    // If divider, return an <hr> element, nothing else
+    if (component === 'hr') {
```

Comment:
use yoda conditions

```suggestion
    if ('hr' === component) {
```

Reply:
ESLint [gets angry](https://eslint.org/docs/latest/rules/yoda) at that one. Can we keep it for now? All the rest of the aybolit code is using normal condition order. I suggest we address this and other code style changes in a large refactor task later, after we deliver the dashboard

Reply:
We do yoda conditions everywhere so either it's incorrectly defined in lint rules or your setup doesn't take into account our config.

Reply:
That's  aybolit/.eslinrc, right? I'll check.

Reply:
Yeah. I checked config and it seems `yoda` is not forced. At the same time it should not get angry and didn't when I ran `npm run lint:js` neither `npx eslint packages/cxl-ui/src/components/cxl-marketing-nav.js`